uint64 timestamp
float64 mag_x
float64 mag_y
float64 mag_z
